sympy/core/tests/test_diff.py

```diff
@@ -109,7 +109,7 @@ def test_diff_nth_derivative():
     f =  Function("f")
     x = Symbol("x")
     y = Symbol("y")
-    n = Symbol("n", integer=True)
+    n = Symbol("n", integer=True, nonnegative=True)
```

Comment:
I wouldn't modify this variable. If the assumption is not specified, I would let the code compute the derivative.